text/0374-jsii-ts-version.md

```diff
@@ -0,0 +1,324 @@
+departure from the [AWS SDKs and Tools maintenance policy][aws-policy].
+
+As such, this warrants ample communication with our customer base to avoid they
+are caught by surprise, or have broken expectations.
+
+### What alternative solutions did you consider?
+
+#### Bring-your-own TypeScript version
+
+One of the main rationales for this change is to enable the use of new
+TypeScript releases without forcing the enitre package ecosystem to make the
+change at the same time (due to language-breaking changes).
+
+An option to address this would be to allow customers to "bring their own"
+TypeScript compiler, by making it a `peerDependency`. This however comes with
+a significant challenge: the TypeScript compiler API is not stable between
+TypeScript releases, and it is hence not possible to author a `jsii` compiler
+that can leverage an arbitrary TypeScript version.
+
+#### Only de-couple the `jsii` package version, but stick with [SemVer]
+
+It would technically be possible to de-couple the `jsii` package from the rest
+of the toolchain's versioning, and to issue a new `jsii` major version each time
+we update the [TypeScript] compiler it internally uses. This would however make
+it difficult for customers to understand the relationship between `jsii` major
+releases and the [TypeScript] language level they support.
+
+Using the same version number prefix (`major.minor`) as the [TypeScript]
+compiler makes the relationship clear and removes the need to maintain separate
+documentation for customers to understand what they are upgrading to.
+
+### What are the drawbacks of this solution?
+
+This solution is a stark departure from versioning schemes and maintenance
+policies used by AWS on open-source products. It implies offering critical bug
+fixes and security updates for up to 5 different release lines (1 current, and
+3 to 4 previous), which might require significant effort.
+
+Following the [TypeScript] `major.minor` reduces our ability to introduce
+breaking changes to the `jsii` compiler, as these must be timed together with
+[TypeScript] `major.minor` release timelines. However, since [TypeScript]
+releases a new `major.minor` line 4 times a year, this is not a blocker, but
+will require careful planning.
+
+### What is the high-level project plan?
+
+Item                                      | Estimation | Notes
+------------------------------------------|------------|--------
+Initial communication                     | 1 day      |
+**Optional:** Break mono-repository out   | 5 days     |
+Have `jsii` down-level `.d.ts`            | 5 days     | [`aws/jsii#3501`][aws/jsii-3501]
+Update release automation                 | 2 days     |
+Update [TypeScript] dependency            | 5 days     |
+Initial Release                           | 1 day      |
+Maintenance Announcement                  | 1 day      |
+Move to Maintenance                       | 1 day      |
+Move to end-of-life                       | 1 day      |
+
```

Comment:
Missing items: updating jsii website documentation, updating jsii contributor guide, updating projen with new defaults for `JsiiProject`, QA / manual validation on some of our larger libraries (construct-hub perhaps?)